cocos/base/CCEventDispatcher.cpp

```diff
@@ -619,7 +619,7 @@ void EventDispatcher::removeEventListener(EventListener* listener)
                 
                 if (_inDispatch == 0)
                 {
-                    listeners->erase(iter);
+                    iter = listeners->erase(iter);
```

Comment:
why do you need the returned value to be stored in iter here ?
Normally, the loop will be broken, then the iter will have no use any more